### PR TITLE
fix(ui5-combobox): Properly focus inner input

### PR DESCRIPTION
**Pull request by d3xter666**
This fix resolves https://github.com/SAP/ui5-webcomponents/issues/2895

### CODE REVIEW

**Comment by d3xter666** [comment]
Hello,

The seems to be failing because of some webdriver issues.

Cheers

**Review by fifoosid — CHANGES_REQUESTED** [review]

**Comment by fifoosid** [comment]
You can merge the master branch into the ```combobox-focus-issue``` branch, because it as about 200 commits behind. This should fix the failing build

File: packages/main/src/ComboBox.js [hunk 1]
```diff
@@ -574,6 +574,10 @@ class ComboBox extends UI5Element {
 		});
 	}
 
+	async focus() {
```
[the remark this answers is not in the document]
Comment (d3xter666) [reply packages/main/src/ComboBox.js:570]: Please check below implementation. It returns this.input.focus() which is async i.e. returns a Promise

**Comment by MapTo0** [comment]
How about adding `data-sap-focus-ref` attribute to the `<input />` element in the `.hbs` file??

**Comment by d3xter666** [comment]
Hi @MapTo0 ,

Can you elaborate more on that `data-sap-focus-ref`?

I tried to replace my implementation by using `data-sap-focus-ref` in the `<input />` field.
However, this does not seem to solve the general issue, described in https://github.com/SAP/ui5-webcomponents/issues/2895

Cheers

**Comment by MapTo0** [comment]
Add a test and we are good to go.

**Comment by d3xter666** [comment]
Tests added

File: packages/main/test/specs/ComboBox.spec.js [hunk 2]
```diff
@@ -351,4 +351,14 @@ describe("Accessibility", () => {
...
 	});
+
+	it ("Should focus the ComboBox with the API", () => {
+		browser.url(`http://localhost:${PORT}/test-resources/pages/ComboBox.html`);
+
+		const combo = $("#combo");
+
+		combo.focus();
+
+		assert.ok(combo["focused"], "ComboBox to be focused");
```
Comment (MapTo0) [inline packages/main/test/specs/ComboBox.spec.js:362]: `combo.getProperty("focused")`

File: packages/main/test/specs/ComboBox.spec.js [hunk 3]
```diff
@@ -351,4 +351,14 @@ describe("Accessibility", () => {
 
 		assert.strictEqual(invisibleMessageSpan.getHTML(false), itemAnnouncement2, "Span value is correct.")
 	});
+
+	it ("Should focus the ComboBox with the API", () => {
+		browser.url(`http://localhost:${PORT}/test-resources/pages/ComboBox.html`);
+
+		const combo = $("#combo");
+
+		combo.focus();
```
Comment (MapTo0) [inline packages/main/test/specs/ComboBox.spec.js:360]: you are interacting with it, not calling focus - click on it or its inner input

Comment (d3xter666) [reply packages/main/test/specs/ComboBox.spec.js]: Done

**Review by MapTo0 — APPROVED** [review]

**Pull request merged**